curp/src/members.rs

```diff
@@ -98,9 +95,15 @@ impl ClusterInfo {
     }
 
     /// Calculate the member id
-    fn calculate_member_id(address: &str, cluster_name: &str, timestamp: Option<u64>) -> ServerId {
+    fn calculate_member_id(
+        addrs: &Vec<String>,
+        cluster_name: &str,
+        timestamp: Option<u64>,
+    ) -> ServerId {
         let mut hasher = DefaultHasher::new();
-        hasher.write(address.as_bytes());
+        for addr in addrs {
+            hasher.write(addr.as_bytes());
+        }
```

Comment:
Before calculating the hash, we need to sort addrs to ensure that the ids calculated by the same addrs are the same.

Reply:
done